src/xdist/scheduler/load.py

```diff
@@ -248,13 +248,19 @@ def schedule(self):
         # Send a batch of tests to run. If we don't have at least two
         # tests per node, we have to send them all so that we can send
         # shutdown signals and get all nodes working.
-        initial_batch = max(len(self.pending) // 4, 2 * len(self.nodes))
-
-        # distribute tests round-robin up to the batch size
-        # (or until we run out)
-        nodes = cycle(self.nodes)
-        for i in range(initial_batch):
-            self._send_tests(next(nodes), 1)
+        if len(self.pending) < 2 * len(self.nodes):
+            # distribute tests round-robin
```

Comment:
Can you please elaborate the comment to explain why we use round-robin for small collections?

Reply:
Honestly, I don't know. To me, round-robin distribution it doesn't look optimal even for small collections. But I just want to stay within "trivial changes and reverts".

I see that 09d79ace356ebf14d221cecd3fd8002d075a2034 improved the case with small number of tests because it distributed tests to all workers, not just first `n / 2`. But that doesn't mean you should do round-robin distribution. I'm just keeping the original code where it works better, trying to achieve the result with minimal changes.

Reply:
> But that doesn't mean you should do round-robin distribution.

Actually you got a point... I can't find the original PR with that change.

@RonnyPfannschmidt do you recall why that was change to use round robin?

Reply:
Maybe it's just simpler to implement than adjusting "sequential" distribution code for the corner case. I've added a comment.

Reply:
PR: https://github.com/pytest-dev/pytest-xdist/pull/13 (commit 09d79ace356ebf14d221cecd3fd8002d075a2034 has a link to it)

Reply:
Thanks! I think one of the reasons is that you have a test module with very slow tests (disregarding fixtures), it is better to distribute that evenly across many workers rather than scheduling most/all of them a single worker. 

I agree with the approach of keeping the changes to a minimal.

Reply:
@nicoddemus Hm... Maybe I chose wrong condition for switching between "old" and "new" algorithms then.

`check_schedule()` has a heuristic for long-running tests. But it will have a chance to rebalance anything only if there are at least 3 tests per node.

Reply:
Hmm I see. Perhaps let's change this condition to `if len(self.pending) < 3 * len(self.nodes):` then?